lib/facter/gnupg_command.rb

```diff
@@ -0,0 +1,18 @@
+# Fact: gnupg_command
+#
+# Purpose: get full path to gpg exe
+#
+# Resolution:
+#   Tests for presence of gpg, returns nil if not present
+#   returns output of which gpg
+#
+# Caveats:
+#   none
+#
+# Notes:
+#   None
+Facter.add(:gnupg_command) do
+  setcode do
+    Facter::Util::Resolution.which('gpg')
```

Comment:
Puppet 3.0.0 depends on Facter >= 1.6.11
https://rubygems.org/gems/puppet/versions/3.0.0

```Facter::Util::Resolution.which``` is present in Facter 1.6.11
https://github.com/puppetlabs/facter/blob/1.6.11/lib/facter/util/resolution.rb#L36

I saw this code from the official puppetlabs java module, after trying a few other ideas.
https://github.com/puppetlabs/puppetlabs-java/blob/master/lib/facter/java_version.rb#L16
I assumed it was compatible with all major versions of puppet supported by that module.